parser_library/src/lsp/lsp_context.cpp

```diff
@@ -768,10 +768,21 @@ std::string lsp_context::get_macro_documentation(const macro_info& m) const
 
 completion_list_s lsp_context::complete_instr(const file_info&, position) const
 {
-    completion_list_s result(completion_item_s::instruction_completion_items_.begin(),
-        completion_item_s::instruction_completion_items_.end());
+    completion_list_s result;
 
-    for (const auto& [_, macro_i] : macros_)
+    // Store only active instructions
```

Comment:
technically, these are active at the end of the open code (we don't keep the whole history), but I guess it should be still more accurate than the state from the beginning.